plugin/nvim-treesitter.vim

```diff
@@ -13,5 +13,20 @@ EOF
 
 let g:loaded_nvim_treesitter = 1
 
+" Text objects
+function! s:text_obj_decl(plug_mapping, default_mapping, funcname)
+  let l:formatted_plug = printf('<Plug>(%s)', a:plug_mapping)
+  if empty(hasmapto(l:formatted_plug))
+    execute printf('vmap %s %s', a:default_mapping, l:formatted_plug)
+    execute printf('omap %s %s', a:default_mapping, l:formatted_plug)
+  endif
+
+  execute printf("vnoremap <silent> %s :<C-U>call %s()<CR>", l:formatted_plug, a:funcname)
+  execute printf("omap <silent> %s :normal v%s<CR>", l:formatted_plug, l:formatted_plug)
+endfunction
+
+call s:text_obj_decl('nvim-treesitter-node', 'grn', 'nvim_treesitter#select_node_incr')
+call s:text_obj_decl('nvim-treesitter-context', 'grc','nvim_treesitter#select_scope_incr')
+
```

Comment:
do we have to use all this vim code ? can't it be scoped to a lua module ?

Reply:
i'm not specially against it, but it would be easier to provide a solid api in lua and then do some quick mappings in vim than splitting the code between vim and lua and trying to use a little bit of both.

Reply:
Strangely yes, because for some reason I was not able to visual select things form lua..

Reply:
That is strange, even with `:normal` ?

Reply:
or `vim.fn.execute` ?

Reply:
also i'm not too sure about setting keymaps, shouldn't we leave that to the users ?

Reply:
the way it behaves is that it provides `<Plug>` mappings and if those are not mapped to anything, then a default mapping is provided for convenience

Reply:
ok so wait for me to finish the configs PR and when it's merged, you can adapt your code to it and we'll merge this